pkg/discovery/backends.go

```diff
@@ -39,6 +40,70 @@ func parse(rawurl string) (string, string) {
 	return parts[0], parts[1]
 }
 
+// ParseAdvertise parses the --cluster-advertise daemon config which accepts
+// <ip-address>:<port> or <interface-name>:<port>
+func ParseAdvertise(store, advertise string) (string, error) {
+	var (
+		iface *net.Interface
+		addrs []net.Addr
+		err   error
+	)
+
+	addrPort := strings.Split(advertise, ":")
+
+	if len(addrPort) > 2 {
+		return "", fmt.Errorf("cluster advertise config (%s) doesnt match <ipv4-address>:<port> format", advertise)
```

Comment:
not sure if this should mention the literal flag name; can this error be triggered in other cases?